### PR TITLE
Make //version not depend on anything.

### PR DESCRIPTION
**Pull request by DarkDimius**
### Motivation
We plan to depend on it using non-C++ languages and want it to have a
more stable ABI

### Test plan

Existing tests

### CODE REVIEW

File: version/version.h [hunk 1]
```diff
@@ -1,23 +1,29 @@
 #ifndef SORBET_VERSION_H
 #define SORBET_VERSION_H
 
-#include <chrono>
```
Comment (DarkDimius) [inline version/version.h]: this file is the main change, everything else is "getting it to work together"

File: common/common.h [hunk 2]
```diff
@@ -31,11 +32,9 @@ template <class E> using UnorderedSet = absl::flat_hash_set<E>;
 // Uncomment to make vectors debuggable
 // template <class T, size_t N> using InlinedVector = std::vector<T>;
 
-#if defined(NDEBUG) && !defined(FORCE_DEBUG)
+#ifdef DEBUG_MODE
 constexpr bool debug_mode = false;
-#undef DEBUG_MODE
 #else
-#define DEBUG_MODE
```
Comment (jez) [inline common/common.h]: This change seems unrelated?

Comment (DarkDimius) [reply common/common.h]: Nope, it's the only part of `common.h` that was fundamentally used in `version.cc`. Now the same macro is defined in `version.c` and consumed in `common.h`, which, TBH, is reasonable, I think

Comment (jez) [reply common/common.h]: In particular, I think this means that`--config=forcedebug` won't work anymore.

Comment (DarkDimius) [reply common/common.h]: > In particular, I think this means that--config=forcedebug won't work anymore.


why so?

https://github.com/sorbet/sorbet/pull/2777/files#diff-6f8df3f313be7fcdd031424e7598b1c7R9

Comment (jez) [reply common/common.h]: Gotcha, didn't see that it had just been moved to version.h :delurk:
